File: Modules/SMES/MES/SIE.Web.MES/Routings/RoutingBoms/Commands/RoutingBomSaveCommand.cs
using SIE.Web.Command;
using System;
using System.Collections.Generic;
using System.Text;

namespace SIE.Web.MES.Routings.RoutingBoms.Commands
{
    /// <summary>
    /// 保存工序BOM主表命令
    /// </summary>
    [JsCommand("SIE.Web.MES.Routings.RoutingBoms.Commands.RoutingBomSaveCommand")]
    public class RoutingBomSaveCommand : FormSaveCommand
    {       
        /// <summary>
        /// 执行命令
        /// </summary>
        /// <param name="args">参数</param>
        /// <param name="scope">作用域</param>
        /// <returns>命令执行结果</returns>
        protected override object Excute(ViewArgs args, string scope)
        {
            return base.Excute(args, scope);
        }
    }

}

File: Modules/SMES/MES/SIE.Web.MES/Routings/RoutingBoms/Commands/RoutingBomSaveCommand.js
SIE.defineCommand('SIE.Web.MES.Routings.RoutingBoms.Commands.RoutingBomSaveCommand', {
    extend: 'SIE.cmd.Save',
    meta: { text: "保存", group: "edit", iconCls: "icon-SaveEntity icon-blue" },
    canExecute: function (view) {
        if (view.getSelection() == null || view.getSelection().length != 1) {
            return false;
        }
        if (view.getCurrent() == null)
            return false;
        if (view.getCurrent()._RoutingBomDetailList == null)
            return true;
        var data = view.getCurrent()._RoutingBomDetailList.data;
        if (data == null || data.length == 0)
            return true;
        return false;
    }
});